services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/stats/MflStatSpec.groovy

```diff
@@ -0,0 +1,134 @@
+
+import groovy.time.TimeCategory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import spock.lang.Narrative
+import spock.lang.Shared
+
+import javax.inject.Provider
+
+@Narrative("""Now we have two FL instances: Management and Statistics.
+- FL Stats: collect statistics only from the switches.
+- FL Management: do the other work and can collect statistics as well when a switch doesn't connect to FL Stats.""")
+class MflStatSpec extends BaseSpecification {
+    @Shared
+    @Value('${opentsdb.metric.prefix}')
+    String metricPrefix
+
+    @Value('${floodlight.controller.management}')
+    private String managementController
+
+    @Value('${floodlight.controller.stat}')
+    private String statController
+
+    @Autowired
+    Provider<TraffExamService> traffExamProvider
+
+    @Tags(VIRTUAL)
+    def "System is able to collect stats from the statistic and management controllers"() {
+        given: "A flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeTraffGens*.switchConnected
+        // not all OVS switches are connected to both controllers by default
+        lockKeeper.setController(srcSwitch.dpId, managementController + " " + statController)
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = 100
+        flowHelper.addFlow(flow)
+
+        when: "Generate traffic on the given flow"
+        def traffExam = traffExamProvider.get()
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildExam(flow, (int) flow.maximumBandwidth)
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stat in openTSDB is created"
+        Date startTime = use(TimeCategory) { new Date() - 20.minutes }
+        def metric = metricPrefix + "flow.raw.bytes"
+        def tags = [switchid: srcSwitch.dpId.toOtsdFormat(), flowid: flow.id]
+        def waitTime = 120
+        def waitInterval = 10
+        def initStat
+        Wrappers.wait(waitTime, waitInterval) {
+            initStat = otsdb.query(startTime, metric, tags).dps
+            assert initStat.size() >= 1
+        }
+
+        when: "Set management controller only on the src switch"
+        lockKeeper.setController(srcSwitch.dpId, managementController)
+
+        and: "Generate traffic on the given flow"
+        exam.setResources(traffExam.startExam(exam, true))
+        assert traffExam.waitExam(exam).hasTraffic()
+
+        then: "Stat on the src switch should be collected because management controller is set"
+        def statFromMgmtController
+        Wrappers.wait(waitTime, waitInterval) {
+            statFromMgmtController = otsdb.query(startTime, metric, tags).dps
+            assert statFromMgmtController.size() > initStat.size()
+            assert statFromMgmtController[statFromMgmtController.keySet()[-2]] <
```

Comment:
`statFromMgmtController.entrySet()[-2].value` is shorter and more readable

Reply:
thx